components/features/Dashboard/Window/Window.tsx

```diff
@@ -37,269 +37,154 @@ const Window: React.FC<Props> = ({
-    setToOriginalSizeMobile,
-  } = useWindowStateManager(
-    app,
-    toggleFullscreen,
-    fullScreenBounds,
-    rndRef.current!
-  );
-
-  const { intersecting, mousePos, mouseDown } = useMouseScreenIntersection();
+  const nodeRef = useRef<HTMLDivElement>(null);
+  const [windowReady, setWindowReady] = useState(false);
+  const [mobileMode, setMobileMode] = useState(false);
+  const [isResizing, setIsResizing] = useState(false);
+  const [mousePosition, setMousePosition] = useState<{
+    x: number | null;
+    y: number | null;
+  }>({ x: null, y: null });
+  const [mouseIntersectingWithTop, setMouseIntersectingWithTop] =
+    useState(false);
   const [dragging, setDragging] = useState(false);
 
-  const topLeftMaximizeArrow: Variants = {
-    hover: {
-      x: -1,
-      y: -1,
-      transition: {
-        repeat: Infinity,
-        repeatType: "reverse",
-        duration: 0.3,
-      },
-    },
-  };
+  useEffect(() => {
+    const snapToCursor = (e: MouseEvent) => {
+      if (dragging && app.fullscreen) {
+        const dragBias = 10;
+        const capturedMousePos = mousePosition;
+        if (
+          Math.abs(e.clientY - capturedMousePos.y!) >= dragBias ||
+          Math.abs(e.clientX - capturedMousePos.x!) >= dragBias
+        ) {
+          const mouseToFullscreenRatio = e.clientX / window.innerWidth;
 
-  const bottomRightMaximizeArrow: Variants = {
-    hover: {
-      x: 1,
-      y: 1,
-      transition: {
-        repeat: Infinity,
-        repeatType: "reverse",
-        duration: 0.3,
-      },
-    },
-  };
+          toggleFullscreen(app, false);
+          onUpdatePos(app, {
+            x: e.clientX - app.width * mouseToFullscreenRatio,
+            y: e.clientY - 15,
+          });
+        }
+      }
+    };
+
+    window.addEventListener("mousemove", snapToCursor);
+
+    return () => window.removeEventListener("mousemove", snapToCursor);
+  }, [dragging, mousePosition, app, onUpdatePos, toggleFullscreen]);
 
-  // this is the reveal 'event' listener that would be fired from the taskbar
   useEffect(() => {
-    const revealSelf = () => {
-      if (bounds && !app.fullscreen) {
-        const newX = (bounds.width - app.width) * 0.5;
-        const newY = (bounds.height - app.height) * 0.5;
-        onUpdatePos(app, {
-          x: newX,
-          y: newY,
-        });
-        rndRef.current?.updatePosition({
-          x: newX,
-          y: newY,
-        });
+    const fullscreenShadow = (e: MouseEvent) => {
+      if (dragging) {
+        if (e.clientY <= 0 && !mouseIntersectingWithTop) {
+          setMousePosition({ x: e.clientX, y: e.clientY });
+          setMouseIntersectingWithTop(true);
+        } else if (mouseIntersectingWithTop && e.clientY >= 0) {
+          setMouseIntersectingWithTop(false);
+        }
       }
     };
 
-    addEventListener("reveal-windows" as keyof WindowEventMap, revealSelf);
+    const killShadow = (e: MouseEvent) => setMouseIntersectingWithTop(false);
```

Comment:
Hello, rashad. I'm kinda curious, if I may ask, is this function required to be defined inside the useEffect? I wonder if it is possible to be defined outside it, maybe inside a useCallback. Let me know 😀

Reply:
Hello Alex, I believe that adding this outside the useEffect will only cause unnecessary complexity to this component. useCallback() should only be used if we plan on reusing the function elsewhere i.e in a child component or in the component itself. The reason I defined inside the useEffect is to also predictably keep the reference of the function inside the effect so the cleanup function dan predictably remove it when the component is unmounted.

Reply:
Hello, rashad. I understood your reasoning, as in keeping the function reference local to only a single render. Thanks for the explanation. 😀